hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java

```diff
@@ -29,14 +29,20 @@
 
 public class ApiTraceGraph {
   private static final Logger LOGGER = LoggerFactory.getLogger(ApiTraceGraph.class);
+
   private static final String UNKNOWN_SPAN_KIND_VALUE =
       EnrichedSpanConstants.getValue(AttributeValue.ATTRIBUTE_VALUE_UNKNOWN);
 
-  private List<ApiNode<Event>> nodeList;
-  private final List<ApiNodeEventEdge> apiNodeEventEdgeList;
   private final StructuredTrace trace;
+  private final List<ApiNodeEventEdge> apiNodeEventEdgeList;
   private final Map<String, Integer> eventIdToIndexInTrace;
 
+  // map of api node to index in api nodes list, since we want to build edges between api nodes
+  private Map<ApiNode<Event>, Integer> apiNodeToIndex;
```

Comment:
I know it was not previously initialized, do you think, we should init in the constructor?